### PR TITLE
Unify(double, TypeReference(struct (0 double))) => TypeReference(struct (0 double)).

### PR DESCRIPTION
**Pull request by ptomin**
- Create Unifier test: Unify(double, TypeReference(struct (0 double))) => TypeReference(struct (0 double))

```
Expected: "STR"
But was:  "(union (real64 u1) (STR u0))"
```
- Unify `(double, TypeReference(struct (0 double))) => TypeReference(struct (0 double))`.

### CODE REVIEW

File: src/Core/Types/Unifier.cs [hunk 1]
```diff
@@ -237,6 +237,25 @@ private bool IsCompatibleWithMemberPointer(MemberPointer mpA, DataType b, int de
             return false;
 		}
 
+        private bool CanBeMergedInto(DataType a, DataType b)
```
Comment (uxmal) [inline src/Core/Types/Unifier.cs:240]: Let's add a comment here like: `Returns true if the data type a can be unified into data type b`

Comment (ptomin) [reply src/Core/Types/Unifier.cs:244]: Done in 4626a2b.

**Comment by ptomin** [comment]
John, I have added requested comment.

**Comment by uxmal** [comment]
It's done.

**Pull request merged**